### DISCUSSION TITLE
Option to preserve whitespace in title

### DISCUSSION THREAD

**Discussion by fisker** [avajs/ava#2928] [General]
In `eslint-plugin-unicorn` project, we use snapshot test, there are lots of js code for the test.

I setup test title to `Invalid #1` `Invalid #2` ... `Invalid #N` at first place, but I found it's hard to locate the failed test code, [so I moved the code into title](https://github.com/sindresorhus/eslint-plugin-unicorn/pull/962). In this way, I can see the testing code in failed test title.

#2779 removes whitespace in AVA 4, we have many tests all about whitespace, [example](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/bd6c16348781d1f35d03deced726ddf7d625f454/test/empty-brace-spaces.mjs#L7).

How can we keep them? Or do you have a better idea?

**Comment by novemberborn** [comment]
Could you share an example where the whitespace normalization breaks your use case? The `cases` you linked to are all on a single line with single spaces so that shouldn't be impacted.

**Reply by fisker** [reply]
https://github.com/sindresorhus/eslint-plugin-unicorn/blob/bd6c16348781d1f35d03deced726ddf7d625f454/test/empty-brace-spaces.mjs#L59

the `/* */` will be replaced with multiple, we also have different indention string in [this file](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/bd6c16348781d1f35d03deced726ddf7d625f454/test/template-indent.mjs), `\t` and `\s` replacement will cause duplicated title.

**Reply by fisker** [reply]
Also, as mentioned, I need display code in title, line breaks removal cause the original code unreadable.

**Reply by novemberborn** [reply]
Your use case is valid, but I think the behavior in AVA 4 is better for more users. I think this is too obscure to have a dedicated setting.

Possible options:

1. A way to, on a per-test basis, avoid normalization. This could be a helper method which tracks strings that must not be normalized.
2. Escape whitespace for test titles (replace `\t` by `\\t` or the `␉` character).
3. Add some other unique value to the titles (an incrementing counter or a hash) so the normalization doesn't matter.

The second option may be the best.
